### DISCUSSION TITLE
About the implementation and performance of the update operation

### DISCUSSION THREAD

**Discussion by Alsace-lee** [milvus-io/milvus#4182] [Q&A and General discussion] [answered]
In the latest scenario, we need to update the 3000w base library with high frequency
The currently adopted method is to execute deleteByEntity() for each batch of updated data, and then execute Insert().
According to the actual measurement of this method, when the amount of data in the base database is large, there will be a performance degradation problem, and it will not respond for a long time.
I want to discuss whether there are similar scenario requirements, and what ways can I consider if I want to achieve it?

在最新的场景中，我们需要对3000w的底库进行高频的更新动作
目前采用的方法是，每批更新数据先执行deleteByEntity()，然后在执行Insert()。
这种方法实测在底库数据量大时，会有性能下降的问题，长时间不能响应。
想讨论下是否有类似的场景需求，如果想要实现可以考虑哪些方式？

**Comment by shiyu22** [comment, marked as answer]
What is your scenario where you need to use frequent data updates?

The current Milvus update scheme only supports to deleteByEntity() and then Insert(), which does affect performance. You can refer to the `Compact segments` in [the storage operation](https://milvus.io/cn/docs/v0.11.0/storage_operation.md), and the attribute filtering is also highly recommended.
